### PR TITLE
Hoisted repeated opening code into private function

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Review by sjarman28 — CHANGES_REQUESTED** [review]
I think moving the device open into its own function is a good change, but there are a couple things that need tidied up. The [device handle declaration](https://github.com/labjack/exodriver/pull/32/commits/615ff744e72af4b8b5c0159f38f7ab431abecc87#diff-00a18286cb6fa29a917374876421cb31f8d78c380b6fa5c9a597a8d7a469ba34R543) in LJUSB_OpenDevice should be removed since that variable is now unused. Additionally, the debug output at [line 580](https://github.com/labjack/exodriver/pull/32/commits/615ff744e72af4b8b5c0159f38f7ab431abecc87#diff-00a18286cb6fa29a917374876421cb31f8d78c380b6fa5c9a597a8d7a469ba34R580)  needs to be conditioned based on whether the handle is NULL.

File: liblabjackusb/labjackusb.c [hunk 1]
```diff
@@ -627,39 +637,7 @@ int LJUSB_OpenAllDevices(HANDLE* devHandles, UINT* productIds, UINT maxDevices)
...
-                if ( r != 0 ) {
-                    fprintf(stderr, "failed to detach from kernel driver. Error Number: %i", r);
-                    libusb_close(devh);
-                    continue;
-                }
-            }
-
-            r = libusb_claim_interface(devh, 0);
-            if (r < 0) {
-                //LJUSB_libusbError(r);
```
Comment (seanm) [inline liblabjackusb/labjackusb.c:656]: @sjarman28 also note a subtle behaviour change here... this commented out `//LJUSB_libusbError(r);` is essentially back.  Hopefully that's ok...

**Comment by seanm** [comment]
@sjarman28 thx. fixed.

File: liblabjackusb/labjackusb.c (hunk 1 above)
Comment (seanm) [reply liblabjackusb/labjackusb.c:656]: thx. fixed.

**Comment by seanm** [comment]
I fixed the unused variable and conditionalized the debug log. thx.

File: liblabjackusb/labjackusb.c (hunk 1 above)
Comment (sjarman28) [reply liblabjackusb/labjackusb.c:656]: I think the behavior change is okay, it should just cause more debug output when debug is enabled.

**Pull request merged**